lib/Raven/ClientBuilder.php

```diff
@@ -201,6 +206,40 @@ public function removeHttpClientPlugin($className)
         return $this;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function addMiddleware(callable $middleware, $priority = 0)
+    {
+        $this->middlewares[] = [$middleware, $priority];
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeMiddleware(callable $middleware)
+    {
+        foreach ($this->middlewares as $key => $value) {
+            if ($value[0] !== $middleware) {
```

Comment:
If you don't find it, you do not complain... Is it the expected behavior?

Reply:
I'm not sure about what to do in case the middleware does not exists. Throwing could be a good idea, but I don't have a strong opinion on the subject. If I would access a specific key of the array and that doesn't exists then throwing an error is ok, in this case it seems that this behavior does not bring any advantage

Reply:
Maybe we could return a bool?

Reply:
Ok

Reply:
I just noticed that while this can be done in the `MiddlewareStack` class, the method in the client builder (which is the one under examination) already returns `$this` to implement the fluent interface